### ISSUE TITLE
addIndex() does not add index?

### ISSUE THREAD

**Issue by MightyPork** [cakephp/phinx#484]
I'll paste my migrations so you can try to reproduce the issue.

To sum it up - I create a table with unique index, and the index is not added (as checked by phpliteadmin).

I have Phinx v0.4.2.1 from composer.

**Migrations**

```
use Phinx\Migration\AbstractMigration;
class Initial extends AbstractMigration
{
    public function change()
    {
        $this->table('articles')
            ->addColumn('canonical',   'text',    ['null' => false])
            ->addColumn('path',        'text',    ['null' => false])
            ->addColumn('hits',        'integer', ['default' => 0])
            ->addColumn('votes_up',    'integer', ['default' => 0])
            ->addColumn('votes_down',  'integer', ['default' => 0])
            ->addIndex('canonical', array('unique' => true))
            ->create();
    }
    public function up() {}
    public function down() {}
}
```

```
use Phinx\Migration\AbstractMigration;
class RemoveVoteColumns extends AbstractMigration
{
    public function up()
    {
        $this->table('articles')
            ->removeColumn('votes_up')
            ->removeColumn('votes_down')
            ->update();
    }
    public function down()
    {
        $this->table('articles')
            ->addColumn('votes_up',    'integer', ['default' => 0])
            ->addColumn('votes_down', 'integer', ['default' => 0])
            ->update();
    }
}
```

```
class AddAliasColumn extends AbstractMigration
{
    public function change()
    {
        $this->table('articles')
            ->addColumn('alias', 'text')
            ->update();
    }
    public function up() {}
    public function down() {}
}
```

```
use Phinx\Migration\AbstractMigration;
class Permalinks extends AbstractMigration
{
    public function change()
    {
        $this->table('permalinks')
            ->addColumn('path',        'text', ['null' => false])
            ->addColumn('canonical',   'text', ['null' => false])
            ->addIndex('path', array('unique' => true))
            ->create();
    }
    public function up() {}
    public function down() {}
}
```

```
use Phinx\Migration\AbstractMigration;
class DropHitsColumn extends AbstractMigration
{
    public function up()
    {
        $this->table('articles')
            ->removeColumn('hits')
            ->update();
    }

    public function down()
    {
        $this->table('articles')
            ->addColumn('hits', 'integer', ['default' => 0])
            ->update();
    }
}
```

```
use Phinx\Migration\AbstractMigration;
class ArticleTableOverhaul extends AbstractMigration
{
    public function change()
    {
        $this->table('articles')
            ->addColumn('name', 'text')
            ->addColumn('description', 'text')
            ->addColumn('url', 'text')
            ->addColumn('created', 'text')
            ->update();
    }
    public function up() {}
    public function down() {}
}
```

I am using it with sqlite database. After migrating, I have index in "permalinks" table, but not in "articles" table.

EDIT: Turns out, the index is deleted just after running the second migration. What's going on??

**Comment by shadowhand** [comment]
Possibly related to #492?

**Comment by MightyPork** [comment]
I'm going to have to replace Phinx with something else, as it really discards indexes on every table modification (it seems, at least).

**Comment by tariquesani** [comment]
Yes, looks like this will be solved if #492 is fixed

**Comment by markstory** [comment]
Closing this as the referenced pull request has been merged for a while now.
